Add deims package

Digdgeo opened the pull request:
Add conda recipe for deims package.

**Package description:**
- deims is a Python package to interface with DEIMS-SDR API
- Used for accessing ecological research site data  
- Version 4.0 from PyPI
- Added original authors as co-maintainers

## Checklist
- [x] Title of this PR is meaningful: "Add deims package"
- [x] License file is packaged (MIT license specified in meta.yaml)
- [x] Source is from official PyPI source
- [x] Package does not vendor other packages
- [x] Build number is 0
- [x] A tarball (PyPI) rather than a repo is used
- [x] GitHub users listed in the maintainer section have been contacted*
- [x] Recipe builds successfully locally

*Note: I am the creator of this recipe with permission from the original authors to create the conda package. Will contact co-maintainers to confirm.

@conda-forge/help-python, ready for review!

synapticarbors reviewed (changes requested):
Please address the comments in https://github.com/conda-forge/staged-recipes/pull/30331#issuecomment-2972899853

https://conda-forge.org/docs/maintainer/knowledge_base/#noarch-python

Digdgeo commented:
Thanks for the review! I've addressed all the suggestions:
- Fixed Python version pins following conda-forge standards
- Updated PyPI URL from pypi.io to pypi.org

Ready for re-review 🚀
